vit-servicing-station-tests/src/common/clients/rest.rs

```diff
@@ -25,37 +65,59 @@ impl RestClient {
     }
 
     pub fn funds(&self) -> Result<Vec<Fund>, RestError> {
-        let content = self.get("funds")?.text()?;
+        let content = self.get_and_verify_status_code("funds")?.text()?;
         if content.is_empty() {
             return Ok(vec![]);
         }
-        serde_json::from_str(&content).map_err(RestError::CannotDeserialize)
+        serde_json::from_str(&content).map_err(|e| RestError::CannotDeserializeResponse {
+            source: e,
+            text: content.clone(),
+        })
     }
 
     pub fn proposals(&self) -> Result<Vec<Proposal>, RestError> {
-        let content = self.get("proposals")?.text()?;
+        let content = self.get_and_verify_status_code("proposals")?.text()?;
+        self.logger.log_text(&content);
         if content.is_empty() {
             return Ok(vec![]);
         }
-        serde_json::from_str(&content).map_err(RestError::CannotDeserialize)
+        serde_json::from_str(&content).map_err(|e| RestError::CannotDeserializeResponse {
+            source: e,
+            text: content.clone(),
+        })
     }
 
     fn get(&self, path: &str) -> Result<reqwest::blocking::Response, reqwest::Error> {
         let request = self.path(path);
-        println!("Request: {}", request);
         let client = reqwest::blocking::Client::new();
         let mut res = client.get(&request);
 
         if let Some(api_token) = &self.api_token {
             res = res.header(API_TOKEN_HEADER, api_token.to_string());
         }
         let response = res.send()?;
-        println!("Response: {:?}", response);
         Ok(response)
     }
 
+    fn get_and_verify_status_code(
+        &self,
+        path: &str,
+    ) -> Result<reqwest::blocking::Response, RestError> {
+        let response = self.get(path)?;
+        if !response.status().is_success() {
+            return Err(RestError::ErrorStatusCode(response.status()));
+        }
+        Ok(response)
+    }
+
+    pub fn disable_log(&mut self) {
+        self.logger.set_enabled(false);
+    }
+
     fn path(&self, path: &str) -> String {
-        format!("http://{}/api/v0/{}", self.address, path)
+        let path = format!("http://{}/api/v0/{}", self.address, path);
```

Comment:
Didn't we have a path builder?

Reply:
In another pr which is not merged yet

Reply:
Cool, we will migrate this piece to the builder later then.